### DISCUSSION TITLE
How to use it as in memory cache back by local file system when data is large?

### DISCUSSION THREAD

**Discussion by marafiq** [microsoft/FASTER#761] [Q&A]
Is this library a good candidate when the use case is: Application has read only data but size of data is large say 10GB. I want to use it as memory cache which is back by file system either local file system or S3?

**Comment by badrishc** [comment]
Yes, this is indeed one of the intended use cases. you can control memory footprint of FASTER by tuning PageSize and MemorySize. Rest of the data will be stored on disk and served up on demand.

**Reply by marafiq** [reply]
Thanks. I see some var length samples using structs as a value object. Do you have any samples where both key and values are c# ref types with variable length?
